projects/cashmere/src/lib/pagination/base-pagination.ts

```diff
@@ -24,17 +26,23 @@ export class BasePaginationComponent extends Initailizable implements OnInit {
     }
     private _length: number = 0;
 
+    private _pageNumber: number = 1;
     /** The currently displayed page. *Defaults to 1.* */
     @Input()
     get pageNumber(): number {
         return this._pageNumber;
     }
     set pageNumber(value: number) {
         const prevPageNumber = this._pageNumber;
-        this._pageNumber = this._sanitizePageNumber(value);
-        this._emitPageEvent(prevPageNumber);
+        this._pageNumber = value;
+
+        const sanitizedValue = this._sanitizePageNumber(value);
+        if (sanitizedValue !== value) {
+            setTimeout(() => (this.pageNumber = sanitizedValue));
```

Comment:
This kills the `ExpressionChangedAfterItHasBeenCheckedError`, by scheduling the value correction for after the current change detection cycle.  See https://blog.angularindepth.com/everything-you-need-to-know-about-the-expressionchangedafterithasbeencheckederror-error-e3fd9ce7dbb4#8c66